torch/_inductor/fx_passes/micro_pipeline_tp.py

```diff
@@ -996,10 +1069,5 @@ def micro_pipeline_tp_pass(graph: torch.fx.Graph):
     for all_gather in all_gathers:
         fuse_all_gather_matmul(all_gather)
 
-    fused_reduce_scatters = False
     for reduce_scatter in reduce_scatters:
-        if fuse_matmul_reduce_scatter(reduce_scatter):
-            fused_reduce_scatters = True
-
-    if reduce_scatters and not fused_reduce_scatters:
-        raise AssertionError("no successful fusions of matul-reduce-scatters")
+        fuse_matmul_reduce_scatter(reduce_scatter)
```

Comment:
Curious why do we decide to change the check of the return value? What if there are no successful fusion occurred, do we still want to raise?

Reply:
Yeah I found that there were only valid scaled_mm-reduce-scatter patterns in the forward graph, but not in the backward graph, so we can't assert this here. In the backward graph, the reduce-scatters are receiving as input the addition of various scaled_mms, so it's not what we are looking to fuse. See diagram below:

![Screenshot 2025-03-28 at 8 41 43 AM](https://github.com/user-attachments/assets/a83f8bb5-e300-4bca-afd4-d7eaa2199143)

Reply:
uh, thanks, his makes sense.